doc/developers-guide/Testing-MongooseIM.md

```diff
@@ -114,31 +132,78 @@ cd $MONGOOSEIM/test/ejabberd_tests/
 
 ```sh
 make quicktest PREPARE=
 ```
 
-Have a look inside the `Makefile` to see how it works.
+Consult the `test.disabled/ejabberd_tests/Makefile` to see how it works.
 
-### Reloading node(s) code
+### Applying code changes
 
-When working on a feature or a bug fix, often you modify the code and check if it works as expected. 
-In order to change the code on dev nodes that are already generated (`mim*` and `fed*`), recompile the code for a specific node.
-For example to update the code on `mim1` node, all you have to do is:
+When working on a feature or a bug fix you often modify the code and check if it works as expected.
+In order to change the code on dev nodes that are already generated (`mim*` and `fed*`) recompile the code for a specific node.
+For example, to update the code on `mim1` node all you have to do is:
 ```sh
 ./rebar3 as mim1 compile
 ```
 
-A similar command applies to other nodes, the important thing here is rebar3's profile.
+A similar command applies to other nodes, the important thing being rebar3's profile.
+
+When the above command finishes the code can be reloaded on the server by either reloading changed module(s) in the node's shell, e.g. `l(mongoose_riak)`, or restarting the node.
+
+### Reading test reports
+
+When finished, test engine writes detailed html reports into a directory:
+
+```
+test.disabled/ejabberd_tests/ct_report/ct_run.[gobbledygook][datetime]/
+```
+
+Each run is saved into a new directory. This snippet:
+
+```
+#!/bin/bash
+
+lst=$(ls -rt ct_report | grep ct_run | tail -n 1)
+rm ct_report/lastrun
+ln -s $lst ct_report/lastrun
+```
+
+can be of some help.
+
+## Checking coverage
+
+If you want to check how much of the code is covered by tests, run:
+
+```
+make cover_quicktest
+```
+
+Remember: to run this, no matter which tests you want to run, you need all mim nodes (mim1, mim2 and mim3) up and running, otherwise test will crash.
+
+This command will recompile and reload code on dev nodes with coverage enabled and run test suites as defined in the spec.
+Coverage statistics will be available in `test.disabled/ejabberd_tests/ct_report/cover.html` and `coverage` subdirectory.
+
+## Advanced topics
+
+There are many more options available.
+One of them is sequentially testing a number of preset configuration - we do it every day on Travis, testing MongooseIM with various OTP versions and database backends.
+Put together, we have eight preset configuration.
+
+If you want to dig deeper, consult `.travis.yml` and `tools/travis-test.sh`, everything we do is there.
+
+## Load testing
+
+Apart from CI, we do also CLT (Continuous Load Testing).
+We have our own load testing infrastructure, called Tide, which is triggered after every successful test run, and gives us a feedback whether we are making MongooseIM faster or slower.
```

Comment:
We have our own load testing infrastructure, called Tide, which is triggered after every successful test run, and gives us a feedback on changes to MongooseIM performance.